src/avrouting.cs

```diff
@@ -0,0 +1,94 @@
+//
+// AVRouting bindings
+//
+// Authors:
+//	TJ Lambert  <TJ.Lambert@microsoft.com>
+//
+// Copyright 2022 Microsoft Corp. All rights reserved.
+//
+
+using System;
+using ObjCRuntime;
+using Foundation;
+using UniformTypeIdentifiers;
+using AVKit;
+using OS_nw_endpoint = System.IntPtr;
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+namespace AVRouting {
+
+	[Mac (13,0), iOS (16,0), MacCatalyst (16,0), NoTV, NoWatch]
+	[Native]
+	public enum AVCustomRoutingEventReason : long
+	{
+		Activate = 0,
+		Deactivate,
+		Reactivate,
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface AVCustomDeviceRoute
+	{
+		[Export ("networkEndpoint")]
+		OS_nw_endpoint NetworkEndpoint { get; }
```

Comment:
This is wrong, we should return the wrapper of the class: https://github.com/xamarin/xamarin-macios/blob/main/src/Network/NWEndpoint.cs#L42

Reply:
Thought about this for a little bit, is this what you mean:

```
		[Internal]
		[Export ("networkEndpoint")]
		OS_nw_endpoint _NetworkEndpoint { get; }

		[Wrap ("NetworkEndpoint (_NetworkEndpoint.GetHandle (), true)")]
		NetworkEndpoint NetworkEndpoint { get; }
```

Reply:
Close, but that is not going to compile, IntPtr does not have a GetHandle() method, and we need to check if it is nullable or not. If it is the IntPtr will be Zero.

Reply:
You might have to surround your code with:

```cs
#if !COREBUILD
...
#endif // !COREBUILD
```

Reply:
@rolfbjarne  That didn't seem to work :/

Reply:
Here is the build logs
https://gist.github.com/tj-devel709/9b74d36290614b2c9f89ed048a0ede71

The first file is when I reference Network.NWEndpoint
and the second file in that gist is when I use "Using Network" at the top of the file

The line 44 is referencing this here inside my avrouting.cs

Reply:
Looks like there are two options:

* Move Network/NWEndpoint.cs in frameworks.sources from the NETWORK_SOURCES variable to a (new) NETWORK_CORE_SOURCES.
* Add `using NWEndpoint = ObjCRuntime.NativeObject;` at the top of this file.

I'd suggest the second approach first, it's easier if it works (the first option will probably snowball into something much bigger, since more files will likely need the same change).

Reply:
Manuel added a commit adding a new AVRouting/AVCustomDeviceRoute.cs to the frameworks.sources where he calls the NWEndpoint constructor so this is resolved now :) Thank you both